src/pke/unittest/utbfvrns/UnitTestBFVrns.cpp

```diff
@@ -0,0 +1,233 @@
+
+//===========================================================================================================
+enum TEST_CASE_TYPE {
+    EVAL_FAST_ROTATION = 0,
+};
+
+static std::ostream& operator<<(std::ostream& os, const TEST_CASE_TYPE& type) {
+    std::string typeName;
+    switch (type) {
+        case EVAL_FAST_ROTATION:
+            typeName = "EVAL_FAST_ROTATION";
+            break;
+        default:
+            typeName = "UNKNOWN";
+            break;
+    }
+    return os << typeName;
+}
+//===========================================================================================================
+struct TEST_CASE_UTBFVRNS {
+    TEST_CASE_TYPE testCaseType;
+    // test case description - MUST BE UNIQUE
+    std::string description;
+
+    UnitTestCCParams params;
+
+    // additional test case data
+    // ........
+
+    std::string buildTestName() const {
+        std::stringstream ss;
+        ss << testCaseType << "_" << description;
+        return ss.str();
+    }
+    std::string toString() const {
+        std::stringstream ss;
+        ss << "testCaseType [" << testCaseType << "], " << params.toString();
+        return ss.str();
+    }
+};
+
+// this lambda provides a name to be printed for every test run by INSTANTIATE_TEST_SUITE_P.
+// the name MUST be constructed from digits, letters and '_' only
+static auto testName = [](const testing::TestParamInfo<TEST_CASE_UTBFVRNS>& test) {
+    return test.param.buildTestName();
+};
+
+static std::ostream& operator<<(std::ostream& os, const TEST_CASE_UTBFVRNS& test) {
+    return os << test.toString();
+}
+//===========================================================================================================
+constexpr usint MULDEPTH = 5;
+constexpr usint PTM      = 65537;
+// clang-format off
+static std::vector<TEST_CASE_UTBFVRNS> testCases = {
+    // TestType,         Descr,  Scheme,        RDim, MultDepth, SModSize, DSize, BatchSz, SecKeyDist, MaxRelinSkDeg, FModSize, SecLvl, KSTech, ScalTech, LDigits, PtMod, StdDev, EvalAddCt, KSCt, MultTech,         EncTech, PREMode
+    { EVAL_FAST_ROTATION, "01", {BFVRNS_SCHEME, DFLT, MULDEPTH,  DFLT,     DFLT,  DFLT,    DFLT,       DFLT,          DFLT,     DFLT,   BV,     DFLT,     DFLT,    PTM,   DFLT,   DFLT,      DFLT, HPSPOVERQLEVELED, DFLT,    DFLT}},
+    { EVAL_FAST_ROTATION, "02", {BFVRNS_SCHEME, DFLT, MULDEPTH,  DFLT,     DFLT,  DFLT,    DFLT,       DFLT,          DFLT,     DFLT,   HYBRID, DFLT,     DFLT,    PTM,   DFLT,   DFLT,      DFLT, HPSPOVERQLEVELED, DFLT,    DFLT}},
+    { EVAL_FAST_ROTATION, "03", {BFVRNS_SCHEME, DFLT, MULDEPTH,  DFLT,     DFLT,  DFLT,    DFLT,       DFLT,          DFLT,     DFLT,   BV,     DFLT,     DFLT,    PTM,   DFLT,   DFLT,      DFLT, HPSPOVERQ, DFLT,    DFLT}},
+    { EVAL_FAST_ROTATION, "04", {BFVRNS_SCHEME, DFLT, MULDEPTH,  DFLT,     DFLT,  DFLT,    DFLT,       DFLT,          DFLT,     DFLT,   HYBRID, DFLT,     DFLT,    PTM,   DFLT,   DFLT,      DFLT, HPSPOVERQ, DFLT,    DFLT}},
```

Comment:
Why are there no unit tests for HPS and BEHZ?

Reply:
Added